tests/mocks/mock_gpu_device_factory.py

```diff
@@ -0,0 +1,35 @@
+# Copyright (c) 2021 NVIDIA CORPORATION. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from unittest.mock import patch, MagicMock, Mock
+from .mock_base import MockBase
+
+
+class MockGPUDeviceFactory(MockBase):
+
+    def __init__(self):
+        self._cuda_visible_gpus = ["GPU_1", "GPU_2"]
+        super().__init__()
+
+    def _fill_patchers(self):
+        self._patchers.append(patch("model_analyzer.device.gpu_device_factory.GPUDeviceFactory.get_cuda_visible_gpus", 
+                                    MagicMock(return_value=dict.fromkeys(self._cuda_visible_gpus))))
+    
+    def get_cuda_visible_gpus(self):
+        return self._cuda_visible_gpus[:]
+    
+    def verify_requested_gpus(self, requested_gpus):
```

Comment:
You can mock this function as well similar to what you did for "model_analyzer.device.gpu_device_factory.GPUDeviceFactory.get_cuda_visible_gpus". If you mock this properly, you can just call `model_analyzer.device.gpu_device_factory.GPUDeviceFactory.verify_requested_gpus` and expect it to run your code.

Reply:
Actually, what I want to do is to fake 2 GPUs using the mock. The `verify_requested_gpus` can be tested using this unit test, so I won't be mocking that function. I removed the `get_cuda_visible_gpus` and `verify_requested_gpus` functions, and just use the expected answer for each test.